### ISSUE TITLE
Undeterministic "Index out of bound" error

### ISSUE THREAD

**Issue by Ant-Bru** [zylo117/Yet-Another-EfficientDet-Pytorch#509]
Hi, I have the following issue performing the training with compound coeff >1:

_**File ".../efficientdet/loss.py", line 105, in forward**_
    **_targets[positive_indices, assigned_annotations[positive_indices, 4].long()] = 1_**
_**IndexError: index 20 is out of bounds for dimension 1 with size 20**_

_**index 20 is out of bounds for dimension 1 with size 20**_

The respective error using GPU is:

_**cuda/IndexKernel.cu:84: operator(): block: [0,0,0], thread: [79,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed.
cuda/IndexKernel.cu:84: operator(): block: [0,0,0], thread: [80,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed.
cuda/IndexKernel.cu:84: operator(): block: [0,0,0], thread: [81,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed.
cuda/IndexKernel.cu:84: operator(): block: [0,0,0], thread: [82,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed.
cuda/IndexKernel.cu:84: operator(): block: [0,0,0], thread: [83,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed.**_

I'm using a custom dataset with 20 classes, having category id starting from 1 (but no problems occur with compound coeff 0 and 1).

**Comment by Ant-Bru** [comment]
I solved the issue, I forgot one class in the configurations (so they are 21) but it's strange that with the wrong config the 0 and 1 versions worked...

**Comment by zylo117** [comment]
No, I don't think it is. If you train long enough, this error should appear again on d0 and d1.

**Comment by hijack007** [comment]
same wrong,did you solve  it ?would you please help me ？
